internal/yaml/yaml.go

```diff
@@ -3,14 +3,49 @@ package yaml
 import (
 	"bufio"
 	"bytes"
+	"io"
 	"os"
 	"strconv"
 	"strings"
 
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	kubeyaml "k8s.io/apimachinery/pkg/util/yaml"
 )
 
+type KubernetesManifestParserFunc func(data []byte) ([]*unstructured.Unstructured, error)
+
+func NewKubernetesManifestParser(scheme *runtime.Scheme) KubernetesManifestParserFunc {
```

Comment:
I think I might prefer to see this somewhere besides the `yaml` package.

For one thing, this handles JSON as well as YAML. If that isn't reason enough, the stuff that's already in this package is much, _much_ lower-level and generic than what you're adding here.

This probably deserves it's own package -- `manifests` maybe.

Then use of the package also looks very tidy -- `manifestParser := manifest.NewParser()` 😄